Remove DATABASE_URL where possible

foarsitter opened the pull request:
This is an attempt the remove `DATABASE_URL` environment variable.

The only use case where `DATABASE_URL` is required is when an application is hosted on heroku. 

Fix #6273 
Closes #6747
Close #6404

foarsitter committed:
Remove DATABASE where possible

browniebroke reviewed (commented):
Looks good in principle, might be nice to see how it looks like after master is merged, as there was a few other changes

foarsitter committed:
Update {{cookiecutter.project_slug}}/config/settings/base.py

Co-authored-by: Bruno Alla <browniebroke@users.noreply.github.com>

foarsitter committed:
Updated with PR feedback

foarsitter commented on docs/developing-locally.rst:
```diff
@@ -55,7 +55,9 @@ First things first.
 
 #. Set the environment variables for your database(s): ::
 
-    $ export DATABASE_URL=postgres://postgres:<password>@127.0.0.1:5432/<DB name given to createdb>
```
The default you proposed for `POSTGRES_HOST` is sufficient I think.

browniebroke committed:
Merge branch 'master' into remove_database_url

# Conflicts:
#	docs/2-local-development/developing-locally.rst
#	{{cookiecutter.project_slug}}/compose/production/django/Dockerfile

browniebroke committed:
Dedent DATABASES dict

browniebroke commented on {{cookiecutter.project_slug}}/.github/workflows/ci.yml:
```diff
@@ -59,8 +59,11 @@ jobs:
+      POSTGRES_USER: 'postgres'
+      POSTGRES_PASSWORD: 'postgres'
+      POSTGRES_DB: 'postgres'
+      POSTGRES_HOST: 'postgres'
```
I _think_ the host used to be `localhost`... Was it intentional or should we keep that?

```suggestion
      POSTGRES_HOST: 'localhost'
```

browniebroke reviewed (commented):
Should we replace this one too?

https://github.com/cookiecutter/cookiecutter-django/blob/575d4c38093a90ec9f311eeb570cdda33121738e/.github/workflows/ci.yml#L97

browniebroke committed:
Merge branch 'main' into remove_database_url

browniebroke committed:
Replace env variables in template CI

browniebroke committed:
Restore DB_URL for runnning compilemessages

browniebroke committed:
Fix generated project GitHub actions

browniebroke committed:
Detect unset vs empty value for DB_URL

browniebroke committed:
Revert "Detect unset vs empty value for DB_URL"

This reverts commit f28dec9f30c78ebd4f76e49c466d95eff6d9ab6a.

browniebroke committed:
Set POSTGRES_HOST on Drone CI config

browniebroke committed:
Add spaces in Jinja tag

browniebroke committed:
Avoid warning from django-environ

.venv/lib/python3.14/site-packages/environ/environ.py:756
  /home/runner/work/django_baked_dburl_2/django_baked_dburl_2/.venv/lib/python3.14/site-packages/environ/environ.py:756: UserWarning: Engine not recognized from url: {'NAME': '', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'ENGINE': ''}
    warnings.warn(f'Engine not recognized from url: {config}')

browniebroke committed:
Fix ruff

browniebroke commented:
@foarsitter I've rebased this and addressed a few issues I found during testing. If you have some time, it would be nice if you could cross check my latest changes.

No worries if you don't have time, though. I'll wait a few days and merge if I don't hear back from you. 

Thanks for getting this started.

browniebroke commented on {{cookiecutter.project_slug}}/config/settings/base.py:
```diff
@@ -48,8 +49,8 @@
 # ------------------------------------------------------------------------------
 # https://docs.djangoproject.com/en/dev/ref/settings/#databases
 
-if db_url := env.db("DATABASE_URL", default=None):
```
With this version, django-environ always tries to parse the URL and logs a warning when it's wrong:

```
environ/environ.py:756: UserWarning: Engine not recognized from url: {'NAME': '', 'USER': '', 'PASSWORD': '', 'HOST': '', 'PORT': '', 'ENGINE': ''}
    warnings.warn(f'Engine not recognized from url: {config}')
```

foarsitter replied:
Good catch :+1:

foarsitter committed:
Merge branch 'main' into remove_database_url
